samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/ExportAction.java

```diff
@@ -0,0 +1,681 @@
+			document.add(shortSummaryTable);
+ 			document.add(new Paragraph(Chunk.NEWLINE));
+
+			shortSummaryTable = new PdfPTable(new float[]{2f, 0.5f, 0.5f, 0.5f});
+
+			this.addCellToTable(shortSummaryTable, rb.getString("short_summary.title"), 1, 0);
+			this.addCellToTable(shortSummaryTable, rb.getString("short_summary.type"), 1, 0);
+			this.addCellToTable(shortSummaryTable, rb.getString("short_summary.answered"), 1, 0);
+			this.addCellToTable(shortSummaryTable, rb.getString("short_summary.score"), 1, 0);
+
+			int index = 0;
+			List<SectionContentsBean> deliveryParts = deliveryBean.getPageContents().getPartsContents();
+			for (SectionContentsBean deliveryPart : deliveryParts) {
+				List<ItemContentsBean> items =	deliveryPart.getItemContents();
+				for (ItemContentsBean item : items) {
+					int tableColor = (index % 2 == 0)? 1 : 2;
+					this.addCellToTable(shortSummaryTable, (++index + ". " + this.cleanText(item.getText())), 2, tableColor);
+					this.addCellToTable(shortSummaryTable, rb.getString("type." + item.getItemData().getTypeId()), 2, tableColor);
+					this.addCellToTable(shortSummaryTable, (!item.isUnanswered() || this.isAnswered(item) ? rb.getString("short_summary.answered.yes") : rb.getString("short_summary.answered.no")), 2, tableColor);
+					this.addCellToTable(shortSummaryTable, (item.getPointsDisplayString() + "/" + item.getMaxPoints()), 2, tableColor);
+				}
+			}
+
+			document.add(shortSummaryTable);
+ 			document.add(new Paragraph(Chunk.NEWLINE));
+			String comments = studentScoreBean.getComments();
+			if (comments != null && !StringUtils.equals(comments, "")) {
+				document.add(new Paragraph(rb.getString("comments_for_student"), boldFont));
+ 				document.add(new Paragraph(Chunk.NEWLINE));
+				PdfPTable commentsTable = new PdfPTable(1);
+				commentsTable.setHorizontalAlignment(PdfPTable.ALIGN_LEFT);
+				this.addCellToTable(commentsTable, comments, 3, 0);
+				document.add(commentsTable);
+			}
+			document.newPage();
+
+			int questionsCuantity = index;
+			int itemsIndex = 0;
+			for (SectionContentsBean deliveryPart : deliveryParts) {
+				List<ItemContentsBean> items =	deliveryPart.getItemContents();
+				for (ItemContentsBean item : items) {
+					Long questionType = item.getItemData().getTypeId();
+					PdfPTable questionTable = new PdfPTable(2);
+					questionTable.setWidthPercentage(50f);
+					questionTable.setHorizontalAlignment(PdfPTable.ALIGN_LEFT);
+					this.addCellToTable(questionTable, ( rb.getFormattedMessage("current_question", new String[]{String.valueOf(++itemsIndex), String.valueOf(questionsCuantity)}) ), 3, 1);
+					this.addCellToTable(questionTable, (item.getPointsDisplayString() + "/" + item.getMaxPoints()), 3, 0);
+					document.add(new Paragraph(Chunk.NEWLINE));
+					document.add(questionTable);
+					if (questionType == TypeIfc.FILL_IN_NUMERIC || questionType == TypeIfc.CALCULATED_QUESTION) {
```

Comment:
https://github.com/sakaiproject/sakai/pull/11809/files#diff-1f00a3cc1fa4cbdc1216b6ac893e483b958e7cb064707d7219f212e821902afeR171-R206

Refactor?

Reply:
Code on lines above are similar except:  item.getFinArray() or item.getFibArray();
Maybe a new method could be created to put all these code only once. 

Something like:  

```
if (questionType == TypeIfc.FILL_IN_NUMERIC || questionType == TypeIfc.CALCULATED_QUESTION) {
  document = processQuestions(document, item.getFibArray())
} else if (questionType == TypeIfc.FILL_IN_BLANK) {
  document = processQuestions(document, item.getFinArray())
}
```

and the processQuestions will contain the same code

```
PdfPTable fillInNumTable = new PdfPTable(1);
fillInNumTable.setHorizontalAlignment(PdfPTable.ALIGN_LEFT);

int i = 0;
String questionText = "";
List<FinBean> fillNumericArray = item.getFinArray();
for (FinBean fillNumericItem : fillNumericArray) {
	if (i + 1 != fillNumericArray.size()) {
		questionText += fillNumericItem.getText() + "(" + (++i) + ")";
		int tableColor = (fillNumericItem.getIsCorrect()!= null && fillNumericItem.getIsCorrect()? 1 : (StringUtils.equals(fillNumericItem.getResponse(), "")? 2 : 3));
		this.addCellToTable(fillInNumTable, (i) + ". " + (StringUtils.equals(fillNumericItem.getResponse(), "")? rb.getString("no_answer.text") : fillNumericItem.getResponse() ), 4, tableColor);
	} else {
		questionText += fillNumericItem.getText();
	}
}
document.add(this.getQuestionTitle(questionText));
document.add(fillInNumTable);
```

It's only an idea...

Reply:
Ok. Thanks, i think it is a good idea, i will aply it